CMake project updates

yeswalrus opened the pull request:
Creating PR to allow comments

mathstuf committed:
protoc: support '=' in --proto_path arguments

seishun committed:
Make implicit defaults consistent with explicit defaults

haberman committed:
Make conformance tests more strict about the failure list.

The failure lists were falling out of date because the
tests would pass even if people forgot to remove failures
from the list.

haberman committed:
Fixed Python by updating failure lists and fixed a few broken tests.

Python 2.x doesn't detect unpaired surrogates so we have to
do that manually.

haberman committed:
Surrogate checking is unpredictable, so always manually check.

haberman committed:
Added update_failure_list.py.

Commit:
Include the prerelease version in the protobuf_VERSION

Commit:
For prerelease versions, require protobuf_FIND_VERSION to be set.

podsvirov committed:
CMake project updates

A series of improvements:

- Improved Protobuf module compatibility (disabled by default);
- Hide advanced settings;
- Added build tree configuration;
- Added build of examples.

yeswalrus commented on cmake/CMakeLists.txt:
```diff
@@ -1,15 +1,18 @@
+
 # Options
 option(protobuf_VERBOSE "Enable for verbose output" OFF)
+option(protobuf_TARGETS_ALIASES "Provide targets aliases without protobuf:: namespace" ON)
```
This should not be allowed, or at the very least should default to off. This has nothing to do with backwards compatibility, and is counter to CMake guidelines.

yeswalrus commented on cmake/CMakeLists.txt:
```diff
@@ -1,15 +1,18 @@
 # Options
 option(protobuf_VERBOSE "Enable for verbose output" OFF)
+option(protobuf_TARGETS_ALIASES "Provide targets aliases without protobuf:: namespace" ON)
+option(protobuf_MODULE_COMPATIBLE "CMake build-in FindProtobuf.cmake module compatible" ON)
```
IMO, if provided, this should default to off. Users can use FindProtobuf.cmake _OR_ the .config files, but they cannot use both. Removing this does not remove a user's option to use the FindProtobuf.cmake file. Making the .config files support the syntax of the FindProtobuf module adds ongoing support burden with little real benefit.

yeswalrus replied:
If it _is_ provided, it should be moved to protobuf-config.in.
config files are only parsed if the version is accepted, so it's perfectly safe and much more helpful to the user.

podsvirov replied:
Ok! We can put the user options in protobuf-options.cmake file and include it in protobuf-config.cmake. Also it can be used in the building of the project, to configure the default behavior.

yeswalrus commented on cmake/install.cmake:
```diff
@@ -6,6 +6,9 @@ foreach(_library
   libprotoc)
   set_property(TARGET ${_library}
     PROPERTY INTERFACE_INCLUDE_DIRECTORIES
+    $<BUILD_INTERFACE:${protobuf_BINARY_DIR}>
```
Given the structure of the cmake files, it seems inappropriate to add these here. 
If a special install.cmake file exists, it implies that it should contain exclusively logic related to installation. Specifying BUILD_INTERFACE rules should happen in the <library>.cmake file.

yeswalrus commented on cmake/install.cmake:
```diff
@@ -88,21 +91,37 @@ else()
 
 configure_file(protobuf-config.cmake.in
-  protobuf-config.cmake @ONLY)
+  ${CMAKE_INSTALL_CMAKEDIR}/protobuf-config.cmake @ONLY)
```
Adding the CMAKE_INSTALL_CMAKEDIR here is counter to standard practices, and really doesn't help anything.

yeswalrus commented on cmake/protobuf-config.cmake.in:
```diff
@@ -1,19 +1,30 @@
 # Version info variables
-set(PROTOBUF_VERSION        "@protobuf_VERSION@")
-set(PROTOBUF_VERSION_STRING "@protobuf_VERSION_STRING@")
+set(Protobuf_VERSION        "@protobuf_VERSION@")
```
${PACKAGE_FIND_NAME}_VERSION is set automatically by find_package if this module is chosen. This line is unnecessary.

yeswalrus commented on cmake/protobuf-config.cmake.in:
```diff
@@ -1,19 +1,30 @@
-set(PROTOBUF_VERSION        "@protobuf_VERSION@")
-set(PROTOBUF_VERSION_STRING "@protobuf_VERSION_STRING@")
+set(Protobuf_VERSION        "@protobuf_VERSION@")
+set(Protobuf_VERSION_STRING "@protobuf_VERSION_STRING@")
```
With the addition of https://github.com/google/protobuf/pull/1624, This becomes a duplicate of the above.

yeswalrus commented on cmake/protobuf-config.cmake.in:
```diff
@@ -1,19 +1,30 @@
-  "${_PROTOBUF_IMPORT_PREFIX}" PATH)
-get_filename_component(_PROTOBUF_IMPORT_PREFIX
-  "${_PROTOBUF_IMPORT_PREFIX}" PATH)
+if(NOT DEFINED protobuf_TARGETS_ALIASES OR "${protobuf_TARGETS_ALIASES}")
```
Again, this is counter to standard cmake practices. Import targets should ALWAYS have a namespace. See [CMP0028](https://cmake.org/cmake/help/v3.5/policy/CMP0028.html)

haberman committed:
Added test for surrogates (valid and invalid).

Commit:
Use ExternalProject_Add to build the examples in a stand-alone fashion.

Commit:
Replace handwritten protobuf-targets with exported version.

haberman committed:
JSON surrogates Python: adjust regex for OSX error message.

OS X's version of Python has a slightly different error
message for this error case.

haberman committed:
Make surrogate regex even more lenient.

haberman committed:
Merge pull request #1541 from haberman/conformancestrict

Make conformance tests more strict about the failure list.

xfxyjwf committed:
Merge pull request #1643 from yeswalrus/cmake-examples

Build examples in CMake

TeBoring committed:
Fix the undefined behavior for opensource users.

xfxyjwf committed:
Merge pull request #1624 from yeswalrus/cmake-prerelease-versioning

CMake prerelease versioning

TeBoring committed:
Add GOOGLE_ prefix before PROTOBUF_DEPRECATED_ATTR

Commit:
Removed handling for ALIASED targets since they are unused.

xfxyjwf committed:
Merge pull request #879 from mathstuf/support-equals-in-proto-path

protoc: support '=' in --proto_path arguments

bshaffer committed:
proper codeblock in README

Commit:
Save the relevant options used to create a package, allow users to reject packages based on them.

Commit:
make protobuf_MSVC_STATIC_RUNTIME a dependent option to reflect it's use.

Commit:
Fix prerelease version matching to be more consistent with the find_package arguments.

xfxyjwf committed:
Merge pull request #1664 from bshaffer/patch-1

proper codeblock in README

sergiocampama committed:
Fixes Xcode 8 analyzer warning saying that it was missing a release in dealloc (#1678)

thomasvl committed:
Add support for generation sources into a framework.

- Add a protoc objc option (generate_for_named_framework) to set the name of
  the framework all generated sources will be in.
- Tweak some comments/naming to make it clear what is the Protobuf framework
  vs. the framework for generated code.
- Update the objc README to document the new generation option to protoc.

This is working towards https://github.com/google/protobuf/issues/1457.

sergiocampama committed:
Add the CocoaPods integration tests to Travis.

Also hotwires updating ruby as CocoaPods was crashing within a support library.

Fixes #1619

thomasvl committed:
Merge pull request #1683 from thomasvl/third_party_framework

Add support for generation sources into a framework.

xfxyjwf committed:
Merge pull request #1665 from yeswalrus/cmake-package-requirements

CMake package requirements

xfxyjwf committed:
Merge pull request #1658 from yeswalrus/cmake-fixup-module

Cleanup protobuf-module.cmake.in

vjpai committed:
Remove a friend-class template that is only used for the
constructor, and instead create an _internal_only getter
that gets the needed information. This is a workaround for
a deficiency in gcc-4.4 that does not properly support
templated friend classes.

haberman committed:
JS: import well-known types from google-protobuf package.

haberman committed:
Bugfix: base the require logic on the file being required.

vjpai committed:
Address review comments on function name

haberman committed:
Added plugin.proto to well-known types for JS.

xfxyjwf committed:
Merge pull request #1692 from vjpai/friendless

Workaround for gcc-4.4 incompatibility

haberman committed:
Fix tests for CommonJS.

thomasvl committed:
Add new generation option for using proto sources from other frameworks.

- Better docs in the generator for the different options that can be passed
  during an invoke of protoc.
- Add named_framework_to_proto_path_mappings_path to pass the path to a file
  containing mappings of frameworks for different proto files.
- Update the generation to use the mapping to change the #import directives
  it creates.

Note: the changes in helpers is mostly moving code within the fine, and then
a small change to expose the parsing so a passed on class can consume the line.

Fixes https://github.com/google/protobuf/issues/1457

haberman committed:
Merge pull request #1696 from haberman/jswkt

JS: import well-known types from google-protobuf package.

xfxyjwf committed:
Merge pull request #1666 from yeswalrus/cmake-prerelease-examples

CMake Fix up prerelease version handling

sergiocampama committed:
Adds destination flag to xcodebuild to avoid possible flake errors (#1697)

Adds destination flag to xcodebuild to avoid possible flake errors

jskeet committed:
Remove ordering guarantees in the MapField documentation

This doesn't currently change the ordering in the implementation, but allows us to do so in the future.
We also need to change
https://developers.google.com/protocol-buffers/docs/reference/csharp-generated#singular
which states "Finally, unlike Dictionary<TKey, TValue>, MapField<TKey, TValue> preserves insertion order of entries."
(We can just remove that sentence, I think.)

thomasvl committed:
Rename methods to avoid ObjC KVC collisions. (#1699)

Note: Breaking API change on the Dictionary classes.

The numeric value classes were using "Value" in the naming, but this silently
collided with the KVC category on NSObject; meaning KVC code could break up a
keypath and call these selectors with the wrong types leading to crashes (even
though the code all would compile cleanly).

- Rename the methods to use the "type" instead of literal "Value".
- Update all the impls and tests.
- Enable the warning that will catch issues like this in the future.

Fixes https://github.com/google/protobuf/issues/1616

jskeet committed:
Merge pull request #1700 from jskeet/ordering

Remove ordering guarantees in the MapField documentation

lukebakken committed:
Modify csharp README since there are now two NuGet packages

lizan committed:
Add JsonParseOptions to ignore unknown fields

- add JsonParseOptions for JsonToBinaryString allow unknown fields
- rename current JsonOptions to JsonPrintOptions

haberman committed:
Create patch release for JS to include WKT.

haberman committed:
Add "google" to package.json "files" for WKT.

haberman committed:
Moved all dependencies to devDependencies.

The runtime doesn't depend on anything, so these
packages should always have been in devDependencies.

jskeet committed:
Expose JsonFormatter.WriteValue.

This isn't useful to most users, but can be handy in advanced use cases,
as requested in #1465.

Commit:
added minified JSON formatting functionality with test

jtattermusch committed:
Merge pull request #1707 from jskeet/format-value

Expose JsonFormatter.WriteValue.

haberman committed:
JS package.json: Added author and updated Closure Library version.

haberman committed:
Merge pull request #1705 from haberman/revjsver

Updates to package.json: rev version, add WKT, and recateogrize dependencies

dkharrat committed:
add nullable qualifier to nil return types

Add the `nullable` qualifier to return types of Objective-C methods that
can return a nil due to errors. This change makes these methods
compatible with the Swift 2 try-catch syntax.

dnkoutso committed:
Get value from text format name in GPBEnumDescriptor

thomasvl committed:
Merge pull request #1714 from dnkoutso/master

Get value from text format name in GPBEnumDescriptor

thomasvl committed:
Don't #import the .m files.

As bazel folks are looking at getting auto generation of module maps going and
the importing of sources files causes issues there.  We were only do it to
hack around some of the apple linker behaviors around objc classes and
categories, but even that isn't complete and CocoaPods was already doing -ObjC,
and developers not using pods could have still needed it to ensure everything
was linked anyways; so drop the hack of importing sources.

esteluk committed:
Fix Objective-C generator option typo

Looks like this was changed during #1683, but missed in the README :)

thomasvl committed:
Merge pull request #1719 from esteluk/patch-1

Fix Objective-C generator option readme typo

thomasvl committed:
Fix GPBGetMessage{Repeated,Map}Field()

- Correct impl by using helpers the message wiring does.
- Add unittests.

Fixes https://github.com/google/protobuf/issues/1716

detlevschwabe committed:
Adding conditional compiler symbol to support .NET 3.5 (#1713)

* Adding condition compiler symbol to support .NET 3.5

thomasvl committed:
Merge pull request #1720 from thomasvl/issue_1716

Fix GPBGetMessage{Repeated,Map}Field()

thomasvl committed:
Xcode project cleanup/setup.

- Correct some cases sources were compiled into the static lib and the tests.
- Enable Xcodes code coverage support on the unittests.  We aren't complete on
  coverage, but having the data always there should make it easier to chip away
  at this going forward.
- Drop method in tests that isn't used, wire up a validator in another test.

thomasvl committed:
Validate the tag numbers when parsing. (#1725)

There was a twist code path (that some times showed up due to what happened to
be in memory in failure cases), that would cast a bogus wire type into the
enum, and then fall through switch statements.

Resolve this by validating all wire types when parsing tags and throwing the
error at that point so it can't enter the system.

As added safety, stick in a few asserts for apis that get passed tags to ensure
they also are only seeing valid data.

Bonus: Tweak the parsing loop to skip some work when we get the end marker
(zero tag) instead of still looping through all the fields.

Commit:
Integrated internal changes from Google

This includes all internal changes from around May 20 to now.

Commit:
Fixed string formatting in text_format.py to be Python2.6-compatible

In Python 2.6 the positional argument specifiers are apparently required
to be explicitly specified:
http://stackoverflow.com/questions/10054122/valueerror-zero-length-field-name-in-format-python

Commit:
Updated failure_list_java.txt to remove tests that now pass

xfxyjwf committed:
Fix windows build.

Change-Id: Ibf7d1df850f4b497303f9f617751be98327898ef

jskeet committed:
Improve exception throwing implementation in collections

jskeet committed:
Implement RepeatedField.AddRange.

This fixes issue #1730.

jskeet committed:
Move DebuggerNonUserCodeAttribute to function members

I think this has caught everything.
I've left a stub for attributes to be applied to the types themselves, but we don't currently need anything.

Follow-up commit will include the changes to generated code itself.

Fixes #1671.

jskeet committed:
Changes to generated code from previous commit

thomasvl committed:
Merge pull request #1723 from thomasvl/objc_test_coverage

Xcode project cleanup/setup.

xfxyjwf committed:
Merge pull request #1710 from chezRong/master

Add minified Json printing functionality

xfxyjwf committed:
Fix some failing travis tests.

1. Add missing header file to Makefile.am.
2. Re-generate objectivec generated code for well-known types.

Change-Id: If28217c701cf8bd739ea0db240e9eee600f23ee7

xfxyjwf committed:
Merge pull request #1738 from xfxyjwf/fixbuild

Fix some failing travis tests.

xfxyjwf committed:
Merge pull request #1704 from lizan/json_parse_options

Add JsonParseOptions to ignore unknown fields

ottok committed:
Fix spelling in strings and comments

ottok committed:
Fix spelling error in function ParseTime parameter

thomasvl committed:
Merge pull request #1712 from dkharrat/swift-error-handling

add nullable qualifier to return types that can be nil, to support Swift 2 try-catch syntax

Commit:
Fixed failing JS tests

A couple of small tweaks were needed to fix test failures that were
inadvertently introduced in the last down integration. I also had to add
the map.js file which was missing.

Commit:
Merge pull request #1752 from acozzette/fix-js-tests

Fixed failing JS tests

xfxyjwf committed:
Don't support global ::string in stringpiece.h

xfxyjwf committed:
Fix data member declaration order.

xfxyjwf committed:
Merge pull request #1742 from ottok/fix-spelling

Fix spelling

jskeet committed:
Remove WriteGeneratedTypeAttributes which is a no-op

This does not affect the generated code.
If we decide we want to apply attributes to generated types, we should start by
just reverting this change.

liujisi committed:
Merge pull request #1753 from xfxyjwf/fixup

Fix problems detected when integrating the code to our internal repo.

dago committed:
Make sure also Solaris x86 gets PATH_MAX

xfxyjwf committed:
Merge pull request #1758 from dago/pathmax2

Make sure also Solaris x86 gets PATH_MAX

jskeet committed:
Merge pull request #1735 from jskeet/attribute-placement

Change placement of DebuggerNonUserCodeAttribute

thomasvl committed:
Merge pull request #1757 from thomasvl/avoid_importing_src

Don't #import the .m files.

thomasvl committed:
Drop the performace baselines.

We weren't really using them, and the nested path causes checkout problems
on windows.

thomasvl committed:
Merge pull request #1762 from thomasvl/drop_perf_profiles

Drop the performace baselines.

jskeet committed:
Optimize AddRange for sequences implementing ICollection

(Also fix a few more C# 6-isms.)

jskeet committed:
Remove unnecessary reflection call

This is the only call to TypeExtensions.IsValueType, so we can remove
that method, making the whole type conditionally compiled out for .NET 3.5

jskeet committed:
Merge pull request #1764 from jskeet/remove-is-value-type

Remove unnecessary reflection call

jskeet committed:
Remove the overload for Add(RepeatedField<T>)

We now just perform the optimization within AddRange itself.

This is a breaking change in terms of "drop in the DLL", but is
source compatible, which should be fine.

jskeet committed:
Merge pull request #1702 from lukebakken/csharp-nuget-doc-update

Modify csharp README since there are now two NuGet packages

Commit:
Fixed out-of-date documentation for CodedInputStream.ReadEnum. (#1581)

thomasvl committed:
Remove the baseline files from the make dist file list.

https://github.com/google/protobuf/pull/1762 remove the files, but I forgot to
update Makefile.am to remove them.

thomasvl committed:
Merge pull request #1776 from thomasvl/fix_dist

Remove the baseline files from the make dist file list.

Commit:
Fix a bad variable dereference causing <package>_FIND_VERSION_PRERELEASE to be ignored.

xfxyjwf committed:
Merge pull request #1778 from yeswalrus/fix-prerelease-version

Fix a <package>_FIND_VERSION_PRERELEASE being ignored

jskeet committed:
Implement RepeatedField.AddRange (#1733)

* Improve exception throwing implementation in collections

* Implement RepeatedField.AddRange.

This fixes issue #1730.

* Optimize AddRange for sequences implementing ICollection

(Also fix a few more C# 6-isms.)

* Remove the overload for Add(RepeatedField<T>)

We now just perform the optimization within AddRange itself.

This is a breaking change in terms of "drop in the DLL", but is
source compatible, which should be fine.

xfxyjwf committed:
Integrate from internal code base.

xfxyjwf committed:
Update generated files.

xfxyjwf committed:
Add missing LIBPROTOBUF_EXPORT.

xfxyjwf committed:
Update version number to 3.0.0-beta-4

xfxyjwf committed:
Add missing LIBPROTOBUF_EXPORT

xfxyjwf committed:
Comment out lite conformance test.

The 'lite' generator flag is no longer supported.

xfxyjwf committed:
Exclude Java lite module from parent pom.xml

liujisi committed:
Merge pull request #1783 from xfxyjwf/fixlite

Comment out lite conformance test.

xfxyjwf committed:
Update version number in AssemblyInfo.cs.

xfxyjwf committed:
Merge pull request #1781 from xfxyjwf/update_version

Update version number to 3.0.0-beta-4

jskeet committed:
Move to dotnet cli for building, and .NET Core (netstandard1.0) as target platform (#1727)

Move to dotnet cli for building, and .NET Core (netstandard1.0) as target platform

This also updates the version number to 3.0.0-beta4

jskeet committed:
Move to dotnet cli for building, and .NET Core (netstandard1.0) as target platform (#1727)

This also updates the version number to 3.0.0-beta4

haberman committed:
Ruby: translate package names from snake_case -> PascalCase.

jskeet committed:
Merge pull request #1785 from jskeet/merge-csharp

Merge C# changes from master to 3.0.0-beta4

xfxyjwf committed:
Add missing golden test file.

xfxyjwf committed:
Fix compatiblity issues.

Currently some public API methods are defined in GenreatedMessage.java
and they have a generric return type:
  class GeneratedMessage {
    class Builder<BuilderType extends Builder<BuilderType>> {
      public BuilderType setField(...);
      public BuilderType setExtension(...);
    }
  }
With these definitions, the compiled byte code of a callsite will have
a direct reference to GeneratedMessage. For example:
  fooBuilder.setField(...);
becomes:
  ##: invokevirtual // Method Builder.setField:(...)LGeneratedMessage.Builder
  ##: checkcast     // class Builder

This will prevent us from updating generated classes to subclass a
different versioned GeneratedMessageV3 class in the future (we can't do
it in a binary compatible way).

This change addresses the problem by overriding these methods directly
in the generated class:
  class Foo {
    class Builder extends GeneratedMessage.Builder<Builder> {
      public Builder setField(...) {
        return super.setField(...);
      }
    }
  }
After this, fooBuilder.setField(...) will be compiled to:
  ##: invokevirtual // Method Builder.setField:(...)LFoo.Builder

The callsites will no longer reference GeneratedMessage directly and we
can change Foo to subclass GeneratedMessageV3 without breaking binary
compatiblity.

The downside of this change is:
  1. It increases generated code size (though it saves some instructions
     on the callsites).
  2. We can never stop generating these overrides because doing that
     will break binary compatibility.

Change-Id: I879afbbc1325a66324a51565e017143489b06e97

xfxyjwf committed:
Merge pull request #1787 from xfxyjwf/steppingstone

Fix compatibility issues for the future GeneratedMessageV3 change.

xfxyjwf committed:
Added 3.0.0-beta-4 changelog.

Change-Id: I997012e7e9b58d9ec8b2f59429d71c98d81aa40d

sergiocampama committed:
Uses head version of rvm to avoid shell_update_session not found error (#1791)

Uses head version of rvm to avoid shell_update_session not found error

Fixes #1786

sergiocampama committed:
Exposes the currently registered extensions for a message and removes the internal sortedExtensionsInUse

xfxyjwf committed:
Mention Java lite in the changelog.

Change-Id: Ic07a7c664930209974244c66885d672288982610

seishun committed:
nits

thomasvl committed:
Use public methods to fetch oneofs in generated code.

When building into frameworks, the generated code doesn't always have direct
access to the proto internals.  Instead of opening up the access, just use the
public method to fetch the correct oneof.

Fixes https://github.com/google/protobuf/issues/1789

thomasvl committed:
Merge pull request #1801 from thomasvl/oneof_framework_build_issues

Use public methods to fetch oneofs in generated code.

xfxyjwf committed:
Remove Java deterministic API.

Change-Id: I43f7e04a53d1445dfa86db310bdb18ceb446398c

xfxyjwf committed:
Merge pull request #1792 from xfxyjwf/changelog

Added 3.0.0-beta-4 changelog.

haberman committed:
JavaScript maps: move binary callbacks out of constructor.

This change will help us separate binary support into
separate files, because we only refer to binary serialization
functions in the actual binary serialization paths.

xfxyjwf committed:
Merge remote-tracking branch 'origin/3.0.0-beta-4'

xfxyjwf committed:
Include javadoc/source in Java release packages.

Also fixed javadoc errors.

[ci skip]

haberman committed:
JavaScript: move extension binary info to separate struct.

xfxyjwf committed:
Merge pull request #1803 from xfxyjwf/javadoc

Include javadoc/source in Java release packages.

haberman committed:
Fix goog.require()/goog.provide() ordering.

haberman committed:
Merge pull request #1802 from haberman/jsmapbin

JavaScript: segregate references to binary functionality

seishun committed:
restore old behavior for toObject

xfxyjwf committed:
Versioning Java GeneratedMessage.

Change-Id: Ib2bb5042deaabdf452d5be2ad1ce40d739ad8d1b

xfxyjwf committed:
Update compatibility tests as well.

Change-Id: I991396ac6e51e32f3ab1daa501d625f34c3ecb04

xfxyjwf committed:
Add files missing from "make dist".

Change-Id: I56a6cce613462794f172ff2e62b25d8a9fc162f3

xfxyjwf committed:
Add missing files in EXTRA_DIST and add a test.

Change-Id: If80725402173cdf50969cb08e7fe5affe3532fb2

jskeet committed:
Use the dotnet-release package feed for Travis.

See https://github.com/dotnet/core/issues/227 for background.

seishun committed:
fix debug.dump

haberman committed:
Merge pull request #1447 from seishun/defaults

JavaScript: Make implicit defaults consistent with explicit defaults

jskeet committed:
Merge pull request #1812 from jskeet/fix-travis

Use the dotnet-release package feed for Travis.

xfxyjwf committed:
Merge pull request #1810 from xfxyjwf/versioning

Versioning Java GeneratedMessage class

xfxyjwf committed:
Merge pull request #1811 from xfxyjwf/fixdist

Add missing files in EXTRA_DIST and add a test.

haberman committed:
Ruby: encode and freeze strings when the are assigned or decoded.

xfxyjwf committed:
Update travis tests for Java.

1. Set JAVA_HOME so mvn can pick up the correct java version.
2. Remove jdk6 tests. It has been broken for a while and remain undetected as
   mvn is actually using java 7 to build the code. Given that we have
   set -source and -target to 6 in the pom.xml and the built .jar
   should be usable by java 6, having a dedicated java 6 test doesn't
   seem necessary (assuming very few Java 6 users want to compile protobuf
   Java from source).

Change-Id: I4f14da772632df3e47801f180198242b306c3f0f

xfxyjwf committed:
Merge pull request #1822 from xfxyjwf/java6

Update travis tests for Java.

xfxyjwf committed:
Add more tests to jenkins.

1. Added ruby22 and jruby tests to jenkins.
2. Added javascript tests to jenkins.
3. Added golang tests to jenkins.
4. Removed ruby19/ruby20 tests from travis. Support for ruby 2.0 has
   ended since 2016/02/24.
   https://www.ruby-lang.org/en/news/2016/02/24/support-plan-of-ruby-2-0-0-and-2-1/

Change-Id: Ie984b06772335352a4be7067ab2485f923875685

sergiocampama committed:
Fixes the parsing of the proto-framework map file.

 - Fixes memory issue where the pointer to the StringPiece would be allocated on the stack, and would mangle the output.
 - Fixes length of the file name when parsing the comma separated files.

xfxyjwf committed:
Merge pull request #1828 from sergiocampama/framework

Fixes the parsing of the proto-framework map file.

xfxyjwf committed:
Merge pull request #1825 from xfxyjwf/jenkins2

Add more tests to jenkins.

xfxyjwf committed:
Add a test to catch sign-comparison warnings.

grpc build treates them as errors and such issues (protobuf change
breaks grpc) has been reported repeatedly. For example:
  https://github.com/google/protobuf/issues/1813

Change-Id: I077c4557cf3effd5195f88802c38999b884edc30

xfxyjwf committed:
Fix sign-comparison warnings in public header files.

xfxyjwf committed:
Remove linux tests from travis.

These tests are already covered by jenkins and right now jenkins can
finish these tests much quicker then travis.

Change-Id: I5799761ccc338211d750940e3cf4c8e2569c8144

xfxyjwf committed:
Keep cpp_distcheck on travis for now.

When it runs in parallel with other jenkins tests, multiple tests start
to fail probably because cpp_distcheck will update the C++ lib that
other tests (such as python_cpp) are depending on.

Change-Id: Ibca0243084bb6b81c811b62c75288e20cc2aabf5

xfxyjwf committed:
Fix maven links.

[ci skip]

Change-Id: Ibefbdf441d932c5751235c635244450d7cbacc53

xfxyjwf committed:
Merge pull request #1829 from xfxyjwf/fixcpp

Fix sign-comparison warnings and add a test for that.

xfxyjwf committed:
Merge pull request #1830 from xfxyjwf/travis

Remove linux tests from travis.

sergiocampama committed:
Correctly sets the generate_for_named_framework option after parsing.

 - Also updates the documentation with a bit more information.

xfxyjwf committed:
Fix maven path.

Change-Id: I252d7a644a9e4064a9f1a816a2be3f25f37b4c43

haberman committed:
Added unit test for PascalCasing package names in Ruby.

haberman committed:
Ruby: fixed string freezing for JRuby.

haberman committed:
Ruby: generated foo.proto -> foo_pb.rb instead of foo.rb.

This brings us more into line with other langauges, and makes it more
obvious when we are requiring protobuf generated code.

haberman committed:
Merge pull request #1821 from haberman/rubyfreezestr

Ruby: encode and freeze strings when the are assigned or decoded.

xfxyjwf committed:
Merge pull request #1834 from sergiocampama/framework

Correctly sets the generate_for_named_framework option after parsing.

xfxyjwf committed:
Merge pull request #1831 from xfxyjwf/protoc

Fix maven links.

haberman committed:
Added new file to ruby_EXTRA_DIST.

xfxyjwf committed:
Add a script to build protoc zip packages.

[ci skip]

Change-Id: Ia0e0bf3bb6bd24633dc8f294318bb20ac837f8b9

haberman committed:
Merge pull request #1837 from haberman/rubygencodename

Ruby: generated foo.proto -> foo_pb.rb instead of foo.rb.

xfxyjwf committed:
Merge pull request #1846 from xfxyjwf/zip

Add a script to build protoc zip packages.

haberman committed:
Merge pull request #1788 from google/rubypackagecap

Ruby: translate package names from snake_case -> PascalCase.

haberman committed:
Fixed Makefile.am for Ruby file rename.

jskeet committed:
Remove legacy_enum_values flag for GA.

jskeet committed:
Bring C#'s ToPascalCase method in line with C++.
(This still doesn't fix the conformance tests, but at least
we're now consistent with the C++ code.)

liujisi committed:
Merge pull request #1859 from jskeet/remove-flag

Remove legacy_enum_values flag for GA.

liujisi committed:
Merge pull request #1861 from jskeet/fix_to_camel_case

Bring C#'s ToPascalCase method in line with C++.

podsvirov committed:
CMake: Link to ZLIB only if protobuf_WITH_ZLIB enabled

podsvirov committed:
CMake: Bugfix for protobuf_MODULE_COMPATIBLE

Removed excess endif()

podsvirov committed:
CMake: Auto find ZLIB from package config if nedded
